test/e2e/cypress/integration/profiles/verification/verify/errors.spec.ts

```diff
@@ -26,120 +26,135 @@ context("Account Verification Error", () => {
-          "contain.text",
-          "An email containing a verification",
-        )
-        cy.get('[name="method"][value="link"]').should("exist")
-        cy.verifyEmailButExpired({ expect: { email: identity.email } })
-      })
+          it("is unable to verify the email address if the code is expired", () => {
+            cy.shortLifespan(s)
+
+            cy.visit(verification)
+            cy.get('input[name="email"]').type(identity.email)
+            cy.get(`button[value="${s}"]`).click()
+
+            cy.get('[data-testid="ui/message/1080001"]').should(
+              "contain.text",
+              "An email containing a verification",
+            )
+            cy.get(`[name="method"][value="${s}"]`).should("exist")
+            cy.verifyEmailButExpired({
+              expect: { email: identity.email },
+              strategy: s,
+            })
+          })
 
-      it("is unable to verify the email address if the code is incorrect", () => {
-        cy.get('input[name="email"]').type(identity.email)
-        cy.get('button[value="link"]').click()
+          it("is unable to verify the email address if the code is incorrect", () => {
+            cy.get('input[name="email"]').type(identity.email)
+            cy.get(`button[value="${s}"]`).click()
 
-        cy.get('[data-testid="ui/message/1080001"]').should(
-          "contain.text",
-          "An email containing a verification",
-        )
+            cy.get('[data-testid="ui/message/1080001"]').should(
+              "contain.text",
+              "An email containing a verification",
+            )
 
-        cy.getMail().then((mail) => {
-          const link = parseHtml(mail.body).querySelector("a")
+            cy.getMail().then((mail) => {
+              const link = parseHtml(mail.body).querySelector("a")
 
-          expect(verifyHrefPattern.test(link.href)).to.be.true
+              expect(verifyHrefPattern.test(link.href)).to.be.true
 
-          cy.visit(link.href + "-not") // add random stuff to the confirm challenge
-          cy.getSession().then(
-            assertVerifiableAddress({
-              isVerified: false,
-              email: identity.email,
-            }),
-          )
-        })
-      })
+              cy.visit(link.href + "-not") // add random stuff to the confirm challenge
+              cy.getSession().then(
+                assertVerifiableAddress({
+                  isVerified: false,
+                  email: identity.email,
+                }),
+              )
+            })
+          })
 
-      it("unable to verify non-existent account", async () => {
-        cy.get('input[name="email"]').type(gen.identity().email)
-        cy.get('button[value="link"]').click()
-        cy.getMail().then((mail) => {
-          expect(mail.subject).eq(
-            "Someone tried to verify this email address (remote)",
-          )
+          it("unable to verify non-existent account", async () => {
```

Comment:
Is this suite missing a test where the code is submitted too often?

Reply:
Good point! Will add it.

Reply:
Done. https://github.com/ory/kratos/pull/2838/commits/51f11c8cda586682ffbef5842c2174186e5d0ac6